### PR TITLE
OSDOCS#9780: Create machine pool (day 2), select a subnet

### PR DESCRIPTION
**Pull request by fmcdonal**
Version(s):
4.14+

Issue:
https://issues.redhat.com/browse/OSDOCS-9780

Link to docs preview:
https://72757--ocpdocs-pr.netlify.app/openshift-rosa/latest/rosa_cluster_admin/rosa_nodes/rosa-managing-worker-nodes#creating_machine_pools_cli_rosa-managing-worker-nodes

QE review:
- [ ] QE has approved this change.

### CODE REVIEW

File: modules/creating-a-machine-pool-cli.adoc [hunk 1]
```diff
@@ -66,6 +67,7 @@ For fault-tolerant worker machine pools, choosing a Multi-AZ machine pool distri
 * A Single-AZ machine pool with one availability zone can have a machine count in multiples of 1, such as 1,2,3,4 and so on.
 ====
 <10> Optional: For machine pools in clusters that do not have Red Hat managed VPCs, you can select additional custom security groups to use in your machine pools. You must have already created the security groups and associated them with the VPC that you selected for this cluster. You cannot add or edit security groups after you create the machine pool. For more information, see the requirements for security groups in the "Additional resources" section.
+<11> Optional: For Multi-AZ clusters and a Single-AZ cluster, you can select subnet to create a Single-AZ machine pool for BYOVPC cluster.
```
Comment (kowen-rh) [inline modules/creating-a-machine-pool-cli.adoc:70]: I feel like this sentence is missing some articles. Do we select _a_ subnet or _the_ subnet? For _a_ BYOVPC cluster or _the_ BYOVPC cluster?

**Review by kowen-rh — COMMENTED** [review]
Seems like we're missing a couple of articles in the callout text.

**Review by oriAdler — APPROVED** [review]

**Review by kowen-rh — APPROVED** [review]
LGTM! :+1:

File: modules/creating-a-machine-pool-cli.adoc [hunk 2]
```diff
@@ -66,6 +67,7 @@ For fault-tolerant worker machine pools, choosing a Multi-AZ machine pool distri
 * A Single-AZ machine pool with one availability zone can have a machine count in multiples of 1, such as 1,2,3,4 and so on.
 ====
 <10> Optional: For machine pools in clusters that do not have Red Hat managed VPCs, you can select additional custom security groups to use in your machine pools. You must have already created the security groups and associated them with the VPC that you selected for this cluster. You cannot add or edit security groups after you create the machine pool. For more information, see the requirements for security groups in the "Additional resources" section.
+<11> Optional: For BYO VPC clusters, you can select a subnet to create a Single-AZ machine pool.
```
Comment (xueli181114) [inline modules/creating-a-machine-pool-cli.adoc:70]: Let's clarify that 
If the subnet is out of cluster creation subnets, there must be a tag "kubernetes.io/cluster/<cluster infra ID>: shared". it is for classic only.
And here is way how to get the cluster infra ID: 
```
lixue@Xue-Lis-MacBook-Pro rosa % rosa describe cluster -c 29rdopsh1f7e2b53gr00o2jlk74cc3pd|grep "Infra ID:" 
Infra ID:                   sdq-ci-weaxr-n8qff
lixue@Xue-Lis-MacBook-Pro rosa %
```

Comment (fmcdonal) [reply modules/creating-a-machine-pool-cli.adoc:70]: @oriAdler can you verify please thanks

Comment (oriAdler) [reply modules/creating-a-machine-pool-cli.adoc:70]: Yes, that's accurate, we have this check in the backend:
checks that the subnet has the tag `kubernetes.io/cluster/<infra-id> : shared`
But I'm not sure if the infra-id is exposed to the user.
At this stage, the preflight fails, and afterward, the user adds the tag and re-attempts.

**Comment by fmcdonal** [comment]
@xueli181114 discussed with @oriAdler and current changes seem accurate as the infra-id is not exposed. Can you confirm you are happy with the new content as is? thanks

**Comment by oriAdler** [comment]
LGTM

**Comment by oriAdler** [comment]
> @xueli181114 discussed with @oriAdler and current changes seem accurate as the infra-id is not exposed. Can you confirm you are happy with the new content as is? thanks

Yes, I assume we need to expose the infra-id in the rosa CLI `describe cluster` command, once we have that we can guide the users to use this value.

File: modules/creating-a-machine-pool-cli.adoc (hunk 2 above)
Comment (xueli181114) [reply modules/creating-a-machine-pool-cli.adoc:70]: @oriAdler Actually this output comes from rosa describe cluster. I mean the `Infra ID` is exposed for classic clusters. Do you mean there is any capability or toggle to control the cluster description?

Comment (oriAdler) [reply modules/creating-a-machine-pool-cli.adoc:70]: My bad, I missed that, thanks for the clarification, if that is the case we can add the comment about the subnet's tag.

**Comment by arendej** [comment]
> Yes, I assume we need to expose the infra-id in the rosa CLI describe cluster command, once we have that we can guide the users to use this value. Alternatively, if the tag is missing, maybe we can prompt a command to the user to add the tag in the ROSA CLI.
> @arendej WDYT?

Yes it would be useful to expose the infra-id. 
In the meantime, calling it out in documentation is necessary. 
@xueli181114 's callout is accurate.

**Comment by fmcdonal** [comment]
@oriAdler and @xueli181114 would you be happy with this new content please.
@arendej are you happy also?

**Comment by oriAdler** [comment]
LGTM

**Pull request merged**